_blueprints/work_item_types.adoc

```diff
@@ -0,0 +1,78 @@
+= Work Items and Work Item Types (Proposal)
+This is a working document to come up with a system for describing and defining types of work items. 
+
+== Vision
+There is a central work item (micro-)service that serves as a common platform for storage and query of work items. Client services extend the basic work item types to offer business functionality like different ways of planning (scrum, waterfall, etc.), testing, ideation, etc. These service in turn are the basis for our UI.
+
+== Goals
+Why do we even have such a thing like a work item type? Why not just store arbitrary json structures? For one, it allows us to optimize storage of work items, for example setting indices on fields in the work items. Work item types also restrict the work items to have a form which we can store efficiently. Work item types can also be queried, as in "give me all work items of type X". This will allow clients to only select the work items they can work with and lets them assume presence of needed fields. 
+
+== Design Principles
+The less privileged components we have, the easier it will be to extend the system (eat our own dogfood). This means keep our core small and use standard extension mechanisms to build advanced functionality.
+The description of a type does not have to be the same as the construction of a type. In constructing work item types, we will probably have some construct of extension or inheritance, but that does not have to show up as a first class construct in the work item type description.

```

Comment:
You mean WorkItemTypes can have a parent/child relationship, but a WorkItem is seen as 'flat' ?